File: SWEA/D2/2005. 파스칼의 삼각형/파스칼의 삼각형.java
//import java.io.FileInputStream;
import java.util.ArrayList;
import java.util.Scanner;

public class Solution {

	public static void main(String[] args) throws Exception {
//		System.setIn(new FileInputStream("src/input.txt"));
		Scanner sc = new Scanner(System.in);
		
		int t = sc.nextInt();
		for (int tc = 1; tc <= t; tc++) {
			int n = sc.nextInt();
			
			ArrayList<Integer>[] dp = new ArrayList[n];
			for (int i = 0; i < n; i++) {
				dp[i] = new ArrayList<Integer>();
				for (int j = 0; j <= i; j++) {
					dp[i].add(1);
				}
			}
			
			for (int i = 2; i < n; i++) {
				for (int j = 1; j < i; j++) {
					dp[i].set(j, dp[i - 1].get(j - 1) + dp[i - 1].get(j));
				}
			}
			
			System.out.printf("#%d%n", tc);
			for (int i = 0; i < n; i++) {
				for (int j = 0; j <= i; j++) {
					System.out.printf("%d ", dp[i].get(j));
				}
				System.out.println();
			}
			
		}
	}

}

File: SWEA/D2/2005. 파스칼의 삼각형/파스칼의 삼각형.py
t = int(input())
for tc in range(1, t + 1):
    n = int(input())
    dp = [[1] * (i + 1) for i in range(n)] # DP 테이블 초기화

    # 값 변경이 필요한 DP 테이블의 행과 열에 대해
    for i in range(2, n):
        for j in range(1, i):
            # 직전 행의 왼쪽 위 열과 바로 위 열 요소 더한 값이
            dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j]  # 현재 행의 열 값

    print(f'#{tc}')
    for i in range(n):
        print(*dp[i])
